sklearn/pipeline.py

```diff
@@ -221,22 +227,58 @@ def _validate_steps(self):
                             "'%s' (type %s) doesn't"
                             % (estimator, type(estimator)))
 
+        return deepcopy(self._steps)
```

Comment:
Yes, or perhaps clone can be modified.

`deepcopy` might be appropriate. Not copying at all might be appropriate. cloning is like deepcopy but it removes fitted models. I'm getting mentally tangled up in what's appropriate for deprecation and what's appropriate in the future.